components/custom_status/date_time_input.tsx

```diff
@@ -0,0 +1,151 @@
+                        onNextClick();
+                    }
+                }}
+            >
+                <svg
+                    width='10'
+                    height='16'
+                    viewBox='0 0 10 16'
+                    fill='none'
+                    xmlns='http://www.w3.org/2000/svg'
+                >
+                    <path
+                        d='M0.41 2.09605L6.314 8.00005L0.41 13.904L1.994 15.506L9.5 8.00005L1.994 0.494048L0.41 2.09605Z'
+                        fill='#3D3C40'
+                        fillOpacity='0.56'
+                    />
+                </svg>
+            </button>
+        </div>
+    );
+};
+
+const DateTimeInputContainer: React.FC = () => {
+    const [day, setDay] = useState('Today');
+    const currentUserId = useSelector(getCurrentUserId);
+    const userTimezone = useSelector((state: GlobalState) => getUserTimezone(state, currentUserId));
+    const locale = useSelector(getCurrentLocale);
+
+    const enableTimezone = useSelector(areTimezonesEnabledAndSupported);
+
+    let currentDate;
+    if (enableTimezone) {
+        if (userTimezone.useAutomaticTimezone) {
+            currentDate = getCurrentDateForTimezone(userTimezone.automaticTimezone);
+        } else {
+            currentDate = getCurrentDateForTimezone(userTimezone.manualTimezone);
+        }
+    } else {
+        currentDate = new Date();
+    }
+
+    const handleDayChange = (day: Date, modifiers: DayModifiers) => {
+        if (modifiers.today) {
+            setDay('Today');
+        } else {
+            setDay(day.toString());
+        }
+    };
+
+    const modifiers = {
+        today: currentDate,
+    };
+
+    return (
+        <div className='dateTime'>
+            <div className='dateTime__date-input'>
+                <span className='dateTime__input-title'>{'Date'}</span>
+                <span className='dateTime__date-icon'>
+                    <i
+                        className='fa fa-calendar'
+                        aria-hidden='true'
+                    />
+                </span>
+                <DayPickerInput
+                    value={day}
+                    onDayChange={handleDayChange}
+                    inputProps={{
+                        readOnly: true,
+                    }}
+                    dayPickerProps={{
+                        navbarElement: <Navbar/>,
+                        fromMonth: currentDate,
+                        modifiers,
+                        locale: locale.toLowerCase(),
+                    }}
+                />
+            </div>
+            <div className='dateTime__time-input'>
+                <span className='dateTime__input-title'>{'Time'}</span>
+                <span className='dateTime__time-icon'>
+                    <i
+                        className='fa fa-clock-o'
+                        aria-hidden='true'
+                    />
+                </span>
+                <input value='12:30 PM'/>
```

Comment:
For the custom date/time field, the next time value should be the time next 30th minute or hour, based on the current time.